CrowdStrike Spotlight c2c doc (#2710)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/crowdStrike-spotlight.md

---
id: crowdstrike-spotlight-source
title: CrowdStrike Spotlight Source
sidebar_label: CrowdStrike Spotlight
description: Learn how to collect combined endpoint vulnerabilities data from the CrowdStrike Spotlight platform.
---
import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/integrations/security-threat-detection/crowdstrike.png')} alt="thumbnail icon" width="85"/>

> The CrowdStrike Spotlight Soucrce will collect CrowdStrike Spotlight combined endpoint vulnerabilities data from the Crowdstrike Falcon instance with Spotlight module enabled. This combined endpoints deliver a unified and comprehensive view of your vulnerability data with a single request.

Reviewer: ```suggestion
The CrowdStrike Spotlight source will collect CrowdStrike Spotlight data combined endpoint vulnerabilities from the CrowdStrike Falcon instance with Spotlight module enabled. This combined endpoints deliver a unified and comprehensive view of your vulnerability data with a single request.
```

## Prerequisites

To collect vulnerability data from the CrowdStrike platform, you must have an authorized CrowdStrike account. CrowdStrike APIs use an OAuth 2.0 authorization token to make authorized API calls. CrowdStrike API Client is required to get the OAuth 2.0 authorization token. To define a CrowdStrike API client, you must be designated as a `CrowdStrike Falcon Administrator` role.

## Data sources

The CrowdStrike Spotlight source ingest vulnerabilities data from the [CrowdStrike Spotlight API](https://falcon.crowdstrike.com/documentation/98/spotlight-apis).

:::important
The CrowdStrike API documentation is not public and can only be accessed by partners or customers.
:::

## Set up and Configuration

> In this configuration, you will set up the CrowdStrike Spotlight and configure it to be authorized and authenticated to use device information from CrowdStrike Spotlight API. To obtain the auth token, you will need the following parameters.

Author: @paulina-kruczek-sumo Thanks! I fixed this.

### API Client and API Secret

> The **API security token** is used to authenticate with CrowdStrike Spotlight API. After successfully creating the API client, you will get the **Client Id**, **Client Secret**, and **Base URL**.
> To get the **CrowdStrike API Client**, follow the steps below:

Reviewer: ```suggestion
The **API security token** is used to authenticate with CrowdStrike Spotlight API. After successfully creating the API client, you will get the **Client Id**, **Client Secret**, and **Base URL**.

To get the **CrowdStrike API Client**, follow the steps below:
```

1. In the **Resources and tools** section, select the **API clients and keys** option. You can then view existing clients or add new API clients from there.<br/><img src={useBaseUrl('img/send-data/crowdstrike-fdr-host-inventory-api-key-client.png')} style={{border: '1px solid black'}} alt="<crowdstrike-spotlight-api-key-client.png>" width="600" />
1. Click **Add new API client**. You will be prompted to give a descriptive name and select the appropriate API scopes.<br/><img src={useBaseUrl('img/send-data/crowdstrike-fdr-host-inventory-new-client.png')} style={{border: '1px solid black'}} alt="<crowdstrike-spotlight-new-client.png>" width="600" />
1. Provide a proper name and description and select the **spotlight-vulnerabilities:read** scope. Click on `ADD` to complete the process.<br/><img src={useBaseUrl('img/send-data/crowdstrike-fdr-host-inventory-add-client.png')} style={{border: '1px solid black'}} alt="<crowdstrike-spotlight-add-client.png>" width="600" />

> 1. After you click on `ADD` a dialogue box will appear with the **Client ID**, **Client Secret** and **Base URL**. Copy and save the Client Id, Client Secret and Base URL to a folder location because you will need them when creating the [CrowdStrike Spotlight Source](#set-up-crowdstrike-spotlight-source).<br/><img src={useBaseUrl('img/send-data/crowdstrike-fdr-host-inventory-copy-creds.png')} style={{border: '1px solid black'}} alt="<crowdstrike-spotlight-copy-creds.png>" width="600" />

Reviewer: ```suggestion
1. After you click on `ADD` a dialogue box will appear with the **Client ID**, **Client Secret** and **Base URL**. Copy and save the Client Id, Client Secret and Base URL to a folder location because you will need them when creating the [CrowdStrike Spotlight source](#set-up-crowdstrike-spotlight-source).<br/><img src={useBaseUrl('img/send-data/crowdstrike-fdr-host-inventory-copy-creds.png')} style={{border: '1px solid black'}} alt="<crowdstrike-spotlight-copy-creds.png>" width="600" />
```

### Region

1. Identify your **Region** based on your **Base URL**. The region can be selected from the list below.

   | Region | Base URL                    |
   | :------ | :-------------------------- |
   | US-1    | https://api.crowdstrike.com |
   | US-2    | https://api.us-2.crowdstrike.com  |
   | EU-1    | https://api.eu-1.crowdstrike.com  |
   | US-GOV-1    | https://api.laggar.gcw.crowdstrike.com |

## States

> A CrowdStrike SpotlightSource tracks errors, reports its health, and start-up progress. You’re informed, in real-time, if the Source is having trouble connecting, if there's an error requiring user action, or if it is healthy and collecting by utilizing Health Events.

Reviewer: ```suggestion
The CrowdStrike Spotlight source tracks errors, reports its health, and start-up progress. You’re informed, in real-time, if the Source is having trouble connecting, if there's an error requiring user action, or if it is healthy and collecting by utilizing Health Events.
```

> A CrowdStrike FSpotlight Source goes through the following states when created:

Reviewer: ```suggestion
The CrowdStrike Spotlight source goes through the following states when created:
```

| FirstPartyGeneric | Normally due to an error communicating with the internal Sumo Logic APIs.                                     | Yes                                                   | The Source will retry indefinitely.                   | FirstPartyGenericError |

### Restarting your Source

{@import ../../../reuse/restart-c2c-source.md}

### JSON configuration

Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [how to use JSON to configure Sources](/docs/send-data/use-json-configure-sources) for details. 

| Parameter | Type | Required | Description | Access |
|:--|:--|:--|:--|:--|
| `config` | JSON Object  | Yes | Contains the [configuration-parameters](#config-parameters) of the Source. | n/a |
| `schemaRef` | JSON Object  | Yes | Use `{"type":"CrowdStrike Spotlight"}` for CCrowdStrike Spotlight Source. | not modifiable |
| `sourceType` | String | Yes | Use `Universal` for CrowdStrike Spotlight Source. | not modifiable |

> ### Config Parameters

Reviewer: ```suggestion
### Config parameters
```
